tests/integration/defs/stress_test/stress_test.py

```diff
@@ -0,0 +1,941 @@
+
+    Args:
+        config: Model configuration for the test (injected by pytest.mark.parametrize)
+        test_mode: Test mode from the --test-mode option ("stress-test" or "stress-stage-alone")
+    """
+    # Test mode handling - determine which tests to run
+    if test_mode == "stress-test":
+        run_performance = True
+        run_stress = True
+    elif test_mode == "stress-stage-alone":
+        run_performance = False
+        run_stress = True
+    else:
+        pytest.skip(
+            f"Skipping test for unsupported mode: {test_mode}. Supported modes: stress-test, stress-stage-alone"
+        )
+        return  # Exit early
+
+    # Skip if not enough GPU memory
+    if get_device_memory() < config.memory_requirement:
+        pytest.skip(
+            f"Not enough GPU memory. Required: {config.memory_requirement}GB")
+
+    # Skip if not enough GPUs for tensor parallelism
+    if get_device_count() < config.tp_size:
+        pytest.skip(f"Not enough GPUs. Required: {config.tp_size}")
+
+    # Get full model path
+    model_path = get_model_path(config.model_dir)
+    model_name = config.model_name
+
+    # Initialize server config that will be used for all tests
+    test_server_config = ServerConfig()
+
+    # Define test configurations
+    performance_config = PerformanceParams() if run_performance else None
+    stress_config = StressTestConfig(
+        model_config=config,
+        server_config=test_server_config) if run_stress else None
+
+    # Check if server is already running
+    is_healthy, _ = check_server_health(test_server_config.url,
+                                        test_server_config.health_check_timeout)
+    if is_healthy:
+        raise RuntimeError(
+            f"Server is already running at {test_server_config.url}. Please stop it manually before running the stress test."
+        )
+
+    # Start server
+    print_info("Starting trtllm-serve server...")
+    print_info(f"Model path: {model_path}")
+
+    # Verify that model path exists
+    if not os.path.exists(model_path):
+        raise RuntimeError(f"Model path does not exist: {model_path}")
+
+    # For tp_size > 1, use multiple GPUs
+    gpu_list = ",".join(str(i) for i in range(config.tp_size))
+
+    # Build server command
+    server_cmd = [
+        "trtllm-serve", model_path, "--port",
+        str(test_server_config.port), "--host", test_server_config.host,
+        "--tp_size",
+        str(config.tp_size)
+    ]
+
+    # Add backend option only if specified
+    # backend = None means trt backend
+    # backend = pytorch means pytorch backend
+    if config.backend:
+        server_cmd.extend(["--backend", config.backend])
+
+    # Log the command we're about to run
+    print_info(f"Running command: {' '.join(server_cmd)}")
+
+    # Start server with the launch_process context manager and filtered output
+    with altered_env(CUDA_VISIBLE_DEVICES=gpu_list):
```

Comment:
If this can be avoided, lets not set this inside the case.

Reply:
Agree, will change